test/e2e/autotls/auto_tls_test.go

```diff
@@ -63,65 +91,51 @@ func TestPerKsvcCertLocalCA(t *testing.T) {
 		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
 	}
 
-	cancel := turnOnAutoTLS(t, clients)
-	test.CleanupOnInterrupt(cancel)
-	defer cancel()
-
-	// Wait for certificate to be ready.
-	if err := waitForCertificateReady(t, clients, routenames.Certificate(objects.Route)); err != nil {
-		t.Fatalf("Error waiting for certificate to become ready: %v", err)
+	certName := getCertificateName(t, clients, objects)
+	if err := waitForCertificateReady(t, clients, certName); err != nil {
+		t.Fatalf("Failed to wait for certificate %s to be ready: %v,", certName, err)
 	}
 
-	// curl HTTPS
-	secretName := routenames.Certificate(objects.Route)
-	rootCAs := createRootCAs(t, clients, objects.Route.Namespace, secretName)
-
 	// The TLS info is added to the ingress after the service is created, that's
-	// why we need to wait again.
+	// why we need to wait again
 	err = v1test.WaitForServiceState(clients.ServingClient, names.Service, v1test.IsServiceReady, "ServiceIsReady")
 	if err != nil {
 		t.Fatalf("Service %s did not become ready: %v", names.Service, err)
 	}
 
+	// curl HTTPS
+	rootCAs := createRootCAs(t, clients, objects.Route.Namespace, certName)
 	httpsClient := createHTTPSClient(t, clients, objects, rootCAs)
 	testingress.RuntimeRequest(t, httpsClient, "https://"+objects.Service.Status.URL.Host)
```

Comment:
shouldn't we be able to check that the status has an HTTPS url and simply use that?

Reply:
`testingress` 👀 this wasn't really intended as a test library :-/